### PR TITLE
Add 'As File' field to the bc edit page

### PR DESCRIPTION
**Pull request by jhadvig**
Adding the field since we can edit it in bc editor.  @spadgett PTAL
screenshot-19

### CODE REVIEW

**Comment by spadgett** [comment]
Missing bindata.

File: assets/app/views/browse/build-config.html [hunk 1]
```diff
@@ -198,6 +198,10 @@
                                 <dt>Output to:</dt>
                                 <dd>{{buildConfig.spec.output.to | imageObjectRef : buildConfig.metadata.namespace}}</dd>
                               </div>
+                              <div ng-if="buildConfig.spec.source.binary">
+                                <dt>As File:</dt>
+                                <dd>{{(buildConfig.spec.source.binary.asFile) ? buildConfig.spec.source.binary.asFile : ""}}</dd>
```
Comment (spadgett) [inline assets/app/views/browse/build-config.html]: We usually show _none_ in italics when there is no value.

```
<dd ng-if="!buildConfig.spec.source.binary.asFile"><em>none</em><</dd>
<dd ng-if="buildConfig.spec.source.binary.asFile">{{buildConfig.spec.source.binary.asFile}}</dd>
```

**Comment by spadgett** [comment]
It's not clear to me looking at the screenshot that this is a binary source. It looks like the result of the build is the file rather than the build input.

/cc @csrwng

**Comment by csrwng** [comment]
not sure how you display the other types of source... but it seems to me that the AsFile field is only additional detail on the Binary source type. So I would expect to see something like:

Source: binary 
and optionally:
Source: binary (as file: blah)

**Comment by spadgett** [comment]
I don't we think we explicitly show the source type on this page. We probably should. Adding @jwforres 

Build details:

https://github.com/openshift/origin/blob/master/assets/app/views/browse/_build-details.html#L37-L55

Build config details:

https://github.com/openshift/origin/blob/master/assets/app/views/browse/build-config.html#L133-L196

We might create directives that we reuse on the build config and build pages so they're always consistent.

**Comment by danmcp** [comment]
@jhadvig Needs to move to the ui repo.

**Pull request closed without merging**